src/databricks/labs/remorph/reconcile/query_builder/expression_generator.py

```diff
@@ -118,6 +125,99 @@ def transform_expression(
     return expr
 
 
+def build_from_clause(table_name: str, table_alias: str | None = None) -> exp.From:
+    return exp.From(this=exp.Table(this=exp.Identifier(this=table_name), alias=table_alias))
+
+
+def build_join_clause(table_name: str, table_alias: str, join_columns: list, kind: str = "inner") -> exp.Join:
+    join_conditions = []
+    for column in join_columns:
+        join_condition = exp.NullSafeEQ(
+            this=exp.Column(this=column, table="source"), expression=exp.Column(this=column, table=table_alias)
+        )
+        join_conditions.append(join_condition)
+
+    # Combine all join conditions with AND
+    on_condition = join_conditions[0]
+    for condition in join_conditions[1:]:
+        on_condition = exp.And(this=on_condition, expression=condition)
+
+    return exp.Join(this=exp.Table(this=exp.Identifier(this=table_name), alias=table_alias), kind=kind, on=on_condition)
+
+
+def build_sub(left_column_name, left_table_name="", right_column_name="", right_table_name="") -> exp.Sub:
+    return exp.Sub(
+        this=build_column(left_column_name, left_table_name),
+        expression=build_column(right_column_name, right_table_name),
+    )
+
+
+def build_threshold_absolute_case(base: exp.Expression, threshold: Thresholds) -> exp.Case:
```

Comment:
this should be inside `threshold_query.py`, since this is specific to threshold as private

Reply:
Done. moved.